nixos/modules/services/cluster/k3s/default.nix

```diff
@@ -33,15 +33,40 @@ in
 
     serverAddr = mkOption {
       type = types.str;
-      description = lib.mdDoc "The k3s server to connect to. This option only makes sense for an agent.";
+      description = lib.mdDoc ''
+        The k3s server to connect to.
+
+        Servers and agents need to communicate each other. Read
+        [the networking docs](https://rancher.com/docs/k3s/latest/en/installation/installation-requirements/#networking)
+        to know how to configure the firewall.
+      '';
       example = "https://10.0.0.10:6443";
       default = "";
     };
 
+    clusterInit = mkOption {
+      type = types.bool;
+      default = cfg.role == "server";
+      defaultText = literalExpression ''cfg.role == "server"'';
+      example = false;
+      description = lib.mdDoc ''
+        Initialize a cluster if k3s is started for the first time.
+
+        If the cluster is already initialized, this option has no effect.
+
+        This option only makes sense in a server that is not connecting to another server.
+
+        If you are configuring an HA cluster, the 1st server must be configured
```

Comment:
"an HA cluster" to "an HA cluster with an embedded DB" maybe? I know the doc comment's also already a bit wordy, but the other type of HA (external datastore) doesn't use the cluster-init flag at all.